userspace/libsinsp/test/public_sinsp_API/interesting_syscalls.cpp

```diff
@@ -305,121 +299,188 @@ TEST(interesting_syscalls, enforce_sinsp_state_with_additions)
+	ASSERT_NAMES_EQ(ordered_syscall_names_final_set, orderd_syscall_names_matching_set);
+	ASSERT_EQ(orderd_syscall_names_matching_set.size(), syscall_names_final_set.size());
+}
 
-	auto final = ordered_syscall_names_final_set.begin();
-	auto matching = orderd_syscall_names_matching_set.begin();
-
-	for(; final != ordered_syscall_names_final_set.end(); final++, matching++)
-	{
-		ASSERT_EQ(*matching, *final);
-	}
+TEST(interesting_syscalls, names_to_sc_set)
+{
+	// "syncfs" is a generic event / syscall
+	static std::set<std::string> orderd_syscall_names = {"kill", "read", "syncfs"};
+	static std::unordered_set<std::string> syscall_names = {"kill", "read", "syncfs"};
+	static libsinsp::events::set<ppm_sc_code> sc_set_matching_set = {PPM_SC_KILL, PPM_SC_READ, PPM_SC_SYNCFS};
+
+	auto sc_set_final = libsinsp::events::names_to_sc_set(syscall_names);
+	ASSERT_PPM_SC_CODES_EQ(sc_set_final, sc_set_matching_set);
+	ASSERT_TRUE(sc_set_final.equals(sc_set_matching_set));
 }
 
-/* This test asserts that `get_events_names` correctly retrieves all the events names
- */
-TEST(interesting_syscalls, get_events_names)
+// TODO add
+TEST(interesting_syscalls, event_set_to_sc_set)
 {
-	std::set<std::string> orderd_events_names_matching_set;
-	libsinsp::events::set<ppm_event_code> events_set;
+		libsinsp::events::set<ppm_sc_code> ppm_sc_set = {
+#ifdef __NR_kill
+	PPM_SC_KILL,
+#endif
 
-	/* Here we don't need ifdefs, our events are always defined. */
-	events_set.insert(PPME_SYSCALL_KILL_E);
-	events_set.insert(PPME_SYSCALL_KILL_X);
-	/* Please note the name of the 2 events should be the same: "kill" */
-	orderd_events_names_matching_set.insert("kill");
+#ifdef __NR_sendto
+	PPM_SC_SENDTO,
+#endif
 
-	events_set.insert(PPME_SYSCALL_DUP_1_E);
-	events_set.insert(PPME_SYSCALL_DUP_1_X);
-	orderd_events_names_matching_set.insert("dup");
+// TODO discuss
+// #ifdef __NR_setresuid32
+	// PPM_SC_SETRESUID32,
+// #endif
+
+// #ifdef __NR_alarm
+// 	PPM_SC_ALARM,
+// #endif
+	};
 
-	auto events_names_final_set = libsinsp::events::event_set_to_names(events_set);
+	libsinsp::events::set<ppm_event_code> event_set = {
+#ifdef __NR_kill
+	PPME_SYSCALL_KILL_E,
+	PPME_SYSCALL_KILL_X,
+#endif
 
-	/* Assert that the 2 sets have the same size */
-	ASSERT_EQ(events_names_final_set.size(), orderd_events_names_matching_set.size());
+#ifdef __NR_sendto
+	PPME_SOCKET_SENDTO_E,
+	PPME_SOCKET_SENDTO_X,
+#endif
 
-	auto ordered_events_names_final_set = test_utils::unordered_set_to_ordered(events_names_final_set);
+// #ifdef __NR_setresuid32
+	// PPME_SYSCALL_SETRESUID_E,
+	// PPME_SYSCALL_SETRESUID_X,
+// #endif
 
-	auto final = ordered_events_names_final_set.begin();
-	auto matching = orderd_events_names_matching_set.begin();
+// #ifdef __NR_alarm
+// 	PPME_GENERIC_E,
+// 	PPME_GENERIC_X,
+// #endif
+	};
 
-	for(; final != ordered_events_names_final_set.end(); final++, matching++)
-	{
-		ASSERT_EQ(*matching, *final);
-	}
+	auto final_sc_set = libsinsp::events::event_set_to_sc_set(event_set);
```

Comment:
Not sure about this test... we will never be able to recover `PPM_SC_ALARM` from `PPME_GENERIC_E/X` . `event_set_to_sc_set` method has not enough info to understand which syscall we want to obtain from the whole generic set...

Reply:
clarified via comment and reduced scope to only test the 1:1 mapping cases since for now we want to keep this API regardless of its limitations.